index.js

```diff
@@ -260,10 +244,23 @@ morgan.token('date', function getDateToken (req, res, format) {
  * response status code
  */
 
-morgan.token('status', function getStatusToken (req, res) {
-  return headersSent(res)
-    ? String(res.statusCode)
-    : undefined
+morgan.token('status', getStatusToken)
+
+/**
+ * colored response status code
+ */
+
+morgan.token('status-colored', function getStatusColoredToken (req, res) {
+  var status = getStatusToken(req, res)
+
+  // get status color
+  var color = status >= 500 ? 31 // red
+    : status >= 400 ? 33 // yellow
+      : status >= 300 ? 36 // cyan
+        : status >= 200 ? 32 // green
+          : 0 // no color
+
+  return color ? '\x1b[' + color + 'm' + status + '\x1b[0m' : status
```

Comment:
I'm commenting here, but this could have just as well been a comment on the docs -- I realize there is a slight quirk in the behavior of this token: technically after the token, it will change the color of everything afterwards as well. For example think of the following format string: `\x1b[47m:method :url :color-status :res[content-length]\x1b[0m`. I believe this token will end up altering the appearance of the `:res[content-length]` when compared to using `:color` token.

I will continue to think about what (if anything) we could do about that, but wanted to comment about this before I completed my thought on it in case you had any thoughts. The reason I mentioned docs is because even if we say there is nothing we can do, it should at least be documented.

Reply:
When I test this implementation I'm only seeing the status get a color.  the `'\x1b[0m'` is an ansi reset code that should reset the color.  It's used in the current `dev` format too: https://github.com/expressjs/morgan/blob/master/index.js#L202

Reply:
Oh, sorry, your terminal may not support background colors. Try this one out: `'\x1b[4m:method :url :color-status :res[content-length]\x1b[0m'`. I can imagine the user would expect everything (but the status) to be blue.

Reply:
yah - in this PR the `:status-colored` token resets any colors at the end so any background color disappears after `:status-colored`.   If the user is setting the background color of their log I think they'd expect that background color for the entire log and have `:status-colored` be whichever color morgan sets it to.

I think the right thing to do is remove the reset at the end of the `:status-colored` token and update `dev` format to add a reset to the end of `:status-colored`.  

That would keep the dev format as is and allow a user to do this if they wanted to add a background color:
`'\x1b[45m:method :url :status-colored \x1b[0m\x1b[45m:res[content-length]\x1b[0m'`

Do you agree?  If so, I'll update the docs

Reply:
Yea, I'm not sure I have a preference with which way to do it, but mainly to call out in the documentation for how folks will need to use the token and what it is doing with the terminal colors.

Reply:
Updated the docs!